pkgs/by-name/op/openfga/package.nix

```
@@ -0,0 +1,27 @@
+{ lib, buildGoModule, fetchFromGitHub, go }:
+
+buildGoModule rec {
+  pname = "openfga";
+  version = "1.5.0";
+
+  src = fetchFromGitHub {
+    owner = "openfga";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-lqAoIOnizfvZm/NjGeinXzoci+ykmzdFtLDGeNoi97c=";
+  };
+
+  vendorHash = "sha256-vUfyo7/mapAzNs6RV+GVj5gOPEpMypNNIkoUgnBYTjA=";
+
+  meta = with lib; {
+    description = "A high performance and flexible authorization/permission engine built for developers and inspired by Google Zanzibar";
+    homepage = "https://openfga.dev/";
+    license = licenses.asl20;
+    mainProgram = "openfga";
+    maintainers = with maintainers; [ mjarosie ];
+  };
+
+  buildPhase = ''
+    go build -o $out/openfga ./cmd/openfga
+  '';
```

```suggestion
  buildPhase = ''
    runHook preInstall
    go build -o $out/openfga ./cmd/openfga
    runHook postInstall
  '';
```

when overriding phases, you need to run their hooks manually.